packages/formatter/src/date.ts

```diff
@@ -265,15 +271,50 @@ function formatDateFriendlyParts(
     dateFormatted = `${parts.weekday.value}, ${parts.month.value}\u00a0${parts.day.value}, ${parts.year.value}`;
   }
 
-  // format the time string
+  const precisionOrder: TimePrecision[] = ['hour', 'minute', 'second'];
+  const maxIndex = precisionOrder.indexOf(maxPrecision);
+  const minIndex = precisionOrder.indexOf(minPrecision);
 
-  let timeFormatted = '';
-  if (parts.minute.value === '00' && parts.second.value === '00') {
-    timeFormatted = `${parts.hour.value}`;
-  } else if (parts.second.value === '00') {
-    timeFormatted = `${parts.hour.value}:${parts.minute.value}`;
-  } else {
-    timeFormatted = `${parts.hour.value}:${parts.minute.value}:${parts.second.value}`;
+  /**
+   *     V min/max > | h | m | s
+   *     h           | X | X | X
+   *     m           | I | X | X
+   *     s           | I | I | X
+   *
+   * I - impossible
```

Comment:
Nit: It wasn't immediately obvious to me what this diagram represents. Maybe a brief explanation like this would help?

This is my reasoning:

Example 1: If max is a larger unit than min, e.g. max = hour, min = minute, then by "min" we must display minute but by max we can't, which is a contradiction.

Example 2: If min is a larger unit than max, e.g. max = minute, min = hour, then by "min" we must display hour and by max we can display at most minute. These do not contradict each other. 

```suggestion
  /**
   *    The maximum precision must be unit larger than or equal to the minimum precision, otherwise the rules will contradict each other. 
   * 
   *     V min/max > | h | m | s
   *     h           | X | X | X
   *     m           | I | X | X
   *     s           | I | I | X
   *
   * X - valid configuration
   * I - invalid configuration
```

Reply:
love it!

Reply:
This comment is actually incorrect, I redid it in d222ef8330f182a47dcbd9ee3207544794ebad31.